ports/esp32/Makefile

```diff
@@ -136,13 +147,28 @@ INC_ESPCOMP += -I$(ESPCOMP)/vfs/include
 INC_ESPCOMP += -I$(ESPCOMP)/xtensa-debug-module/include
 INC_ESPCOMP += -I$(ESPCOMP)/wpa_supplicant/include
 INC_ESPCOMP += -I$(ESPCOMP)/wpa_supplicant/port/include
-INC_ESPCOMP += -I$(ESPCOMP)/ethernet/include
 INC_ESPCOMP += -I$(ESPCOMP)/app_trace/include
 INC_ESPCOMP += -I$(ESPCOMP)/app_update/include
 INC_ESPCOMP += -I$(ESPCOMP)/pthread/include
 INC_ESPCOMP += -I$(ESPCOMP)/smartconfig_ack/include
 INC_ESPCOMP += -I$(ESPCOMP)/sdmmc/include
 
+ifeq ($(ESPIDF_CURHASH),$(ESPIDF_SUPHASH_V4))
+INC_ESPCOMP += -I$(ESPCOMP)/esp_eth/include
+INC_ESPCOMP += -I$(ESPCOMP)/esp_event/private_include
+INC_ESPCOMP += -I$(ESPCOMP)/lwip/include/apps/sntp
+INC_ESPCOMP += -I$(ESPCOMP)/wpa_supplicant/include/esp_supplicant
+INC_ESPCOMP += -I$(ESPCOMP)/spi_flash/private_include
+INC_ESPCOMP += -I$(ESPCOMP)/xtensa/include
+INC_ESPCOMP += -I$(ESPCOMP)/xtensa/esp32/include
+INC_ESPCOMP += -I$(ESPCOMP)/esp_rom/include
+INC_ESPCOMP += -I$(ESPCOMP)/esp_common/include
+INC_ESPCOMP += -I$(ESPCOMP)/esp_wifi/include
+INC_ESPCOMP += -I$(ESPCOMP)/esp_wifi/esp32/include
+else
+INC_ESPCOMP += -I$(ESPCOMP)/ethernet/include
+endif
```

Comment:
The expat, micro-ecc and nghttp lines (5 in total from above) will also need to be moved here.